tests/unit/zhmcclient/test_manager.py

```diff
@@ -820,6 +822,214 @@ def test_update_populated_modify(self):
 
         # Access the new entry, and check the entry (and that list() has not
         # been called again).
-        act_resource2_uri = self.cache.get(self.resource2_name)
+        act_resource2_name, act_resource2_uri = \
+            self.cache.get(self.resource2_name)
         assert self.manager._list_called == 1
         assert act_resource2_uri == resource2_new_uri
+        assert act_resource2_name == self.resource2_name
+
+
+TESTCASES_FINDALL_NAME_PARMS = [
+    # Testcases for test_findall_name_parms().
+    # Each list item is a tuple defining a testcase in the following format:
+    # - res_names: List of names of existing resources
+    # - name_filter: Filter value for matching the resource name
+    # - exp_names: Expected resource names in result
+    # - exp_exc: Expected exception, or None
+    (
+        ['name1', 'name2'],
+        'name1',
+        ['name1'],
+        None
+    ),
+    (
+        ['name1', 'name2'],
+        'name.*',
+        ['name1', 'name2'],
+        None
+    ),
+    (
+        ['name1', 'name2'],
+        'ame',
+        [],
+        None
+    ),
+    (
+        ['name-1', 'name.1', 'name1'],
+        'name.1',
+        ['name-1', 'name.1'],
+        None
+    ),
+    (
+        ['name-1', 'name.1', 'name1'],
+        r'name\.1',
+        ['name.1'],
+        None
+    ),
+    (
+        ['nax', 'nam', 'name', 'name1'],
+        'name?',
+        ['nam', 'name'],
+        None
+    ),
+    (
+        ['nax', 'nam', 'name', 'namee', 'name1'],
+        'name*',
+        ['nam', 'name', 'namee'],
+        None
+    ),
+    (
+        ['nax', 'nam', 'name', 'namee', 'name1'],
+        'name+',
+        ['name', 'namee'],
+        None
+    ),
+]
+
+
+@pytest.mark.parametrize(
+    "res_names, name_filter, exp_names, exp_exc",
+    TESTCASES_FINDALL_NAME_PARMS)
+def test_findall_name_parms(res_names, name_filter, exp_names, exp_exc):
+    """
+    Test BaseManager.findall() with filter args for matching the resource name.
+    """
+
+    session = Session(host='fake-host', userid='fake-user', password='fake-pw')
+    manager = MyManager(session)
+    all_resources = []
+    for res_name in res_names:
+        uri = res_name + '-uri'
+        res = MyResource(
+            manager,
+            uri=uri,
+            properties={
+                manager._name_prop: res_name,
+                "other": "fake-other",
+                "same": "fake-same",
+                "int_other": 23,
+                "int_same": 42,
+            })
+        all_resources.append(res)
+    # pylint: disable=attribute-defined-outside-init
+    manager._list_resources = all_resources
+
+    filter_args = {manager._name_prop: name_filter}
+
+    if exp_exc:
+        with pytest.raises(exp_exc):
+
```

Comment:
is there a special reason for those empty lines around here? Looks weird.

Reply:
I want the code to be tested surrounded by empty lines so that it cannot be overlooked in the debugger. The empty line before the if is because this is logically a new section.